### PR TITLE
Big upgrade

### PR DESCRIPTION
**Pull request by gchazot**
Upgrading to latest versions of dependencies, getting it to work *fast* with Python3 and adding some integration tests.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,15 +1,27 @@
...
 ]
 
+# Regex matching version pattern
+# (3 numerical values separated by `.`, semver style, followed by an optional pre-release marker)
+version_pattern = re.compile(r"\d+\.\d+\.\d+([.-][\w_-]+)?")
+
+
+def get_version():
+    with open("CHANGELOG.md", "r") as fn:
+        while True:
```
Comment (sujaymansingh) [inline setup.py]: Not a major thing, but if there is no line that matches the regex, won't this loop forever?
Any reason not to iterate over the lines in the file instead?

File: setup.py [hunk 2]
```diff
@@ -19,7 +31,7 @@
 
     setuptools.setup(
         name="queue_util",
-        version="2.2.1",
+        version=get_version(),
         author="Sujay Mansingh",
         author_email="sujay.mansingh@gmail.com",
```
Comment (sujaymansingh) [inline setup.py]: Might be worth changing the author and author email...

File: setup.py (hunk 1 above)
Comment (gchazot) [reply setup.py]: Good catch @sujaymansingh. Copy-Pasta gone ~wrong~ as usual

Comment (gchazot) [reply setup.py]: 28c5ba7

File: setup.py (hunk 2 above)
Comment (gchazot) [reply setup.py]: 43a1b69

**Comment by gchazot** [comment]
Thanks for the reviews @Fizzadar @sujaymansingh

**Pull request merged**